src/Turtle/Line.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE DeriveDataTypeable         #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Turtle.Line
+  ( Line(lineToText)
+  , textToLines
+  , linesToText
+  , textToLine
+  , unsafeTextToLine
+  , NewlineForbidden(..)
+  ) where
+
+import Data.Text (Text)
+import qualified Data.Text as Text
+#if __GLASGOW_HASKELL__ >= 708
+import Data.Coerce
+#endif
+import Data.String
+#if __GLASGOW_HASKELL__ >= 710
+#else
+import Data.Monoid
+#endif
+import Data.Maybe
+import Data.Typeable
+import Control.Exception
+
+data NewlineForbidden = NewlineForbidden
```

Comment:
I added a comment that explains what may cause the exception and how to debug it.